### PR TITLE
Feature/ll adaptivecards

### PR DESCRIPTION
**Pull request by llakewood**
# Description

Adding in Microsoft [AdaptiveCards](https://adaptivecards.io/explorer/) framework to solve design problems like longer text strings, adding links, creating bullet points, and embedding images.

## Test Instructions

1. Start bot
2. Select 'yes' from unblock prompt
3. Observe the text and image formats in the following text prompts
4. Complete user input of direct deposit info as prompted
-----
5. Open code editor and change src/dialogs/unblockDialogs/unblockDirectDeposit.ts 
const CARD_DEBUG = null; to = true;
6. Restart the Bot, see the alternate approach to the same flow

## Help Requested

- Unit testing framework

## Checklist
- [x] Update CHANGELOG.md
- [ ] Unit tests have been added/updated
- [ ] e2e tests have been added/updated

### CODE REVIEW

**Review by binggaofw — COMMENTED** [review]
I think better to move all LUIS setup code to a separate class and call it in dialog. otherwise you will need to copy and paste LUISAppSetup in every step that need using luis service

**Review by binggaofw — COMMENTED** [review]
confirmAuthWordStep.ts and confirmCallbackDetailsStep.ts and getPreferredCallbackDateAndTimeStep.ts can be deleted

**Review by binggaofw — COMMENTED** [review]
ConfirmHomeAddressStep better change step name to some meaningful name ,not just named as first/second/thirdStep etc

File: src/dialogs/confirmCallbackStep.ts [hunk 1]
```diff
@@ -45,10 +44,12 @@ export class ConfirmCallbackStep extends ComponentDialog {
     let okMsg;
     let standardMsg;
 
+    console.log(stepContext)
+
     switch(stepContext.stack[0].id){
       case 'CONFIRM_CALLBACK_WATERFALL_STEP':
         okMsg = false;
-        standardMsg = i18n.__('confirmHomeAddressFailedErrorMsg');
+        standardMsg = i18n.__('unblock_direct_deposit_main_error');
```
Comment (binggaofw) [inline src/dialogs/confirmCallbackStep.ts:52]: not sure why here change the standard message

File: src/dialogs/unblockDialogs/confirmHomeAddressStep.ts [hunk 2]
```diff
@@ -44,15 +44,15 @@ const LUISAppSetup = (stepContext) => {
   // LUIZ Recogniser processing
```
Comment (binggaofw) [inline src/dialogs/unblockDialogs/confirmHomeAddressStep.ts:44]: better create a luis class to handle this so you do not need put this in every step

File: src/dialogs/unblockDialogs/unblockDirectDeposit.ts [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+  async unblockDirectDepositStart(stepContext) {
+
+    // Get the user details / state machine
+    const unblockBotDetails = stepContext.options;
+
+    // Check if the error count is greater than the max threshold
+    if (unblockBotDetails.errorCount.unblockDirectDeposit >= MAX_ERROR_COUNT) {
+
+      unblockBotDetails.masterError = true;
+      unblockBotDetails.unblockDirectDeposit = null;
```
Comment (binggaofw) [inline src/dialogs/unblockDialogs/unblockDirectDeposit.ts:58]: if this unblockDirectDeposit reach max error count, unblockBotDetails.unblockDirectDeposit set to -1 is better i think

File: src/dialogs/unblockDialogs/unblockDirectDeposit.ts [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+        new CallbackBotDetails()
+      );
+    }
+
+    // If it is in the error state (-1) or or is set to null prompt the user
+    // If it is false the user does not want to proceed
+    if (
+      unblockBotDetails.unblockDirectDeposit === null ||
+      unblockBotDetails.unblockDirectDeposit === -1 ||
+      unblockBotDetails.unblockDirectDeposit === 0
```
Comment (binggaofw) [inline src/dialogs/unblockDialogs/unblockDirectDeposit.ts:71]: not sure the 0 means here

**Comment by binggaofw** [comment]
i think overall very good, just some small parts need to twist. we can discuss tomorrow

File: src/dialogs/confirmCallbackStep.ts (hunk 1 above)
Comment (llakewood) [reply src/dialogs/confirmCallbackStep.ts:52]: this is a portion I need help with, i'll explain in a different PR. This is a TODO.

**Comment by llakewood** [comment]
Closing this PR as I've made these changes in #8.

**Pull request closed without merging**